### ISSUE TITLE
Regression: Retry executeCheckedSupplier() throws NPE on Interrupt

### ISSUE THREAD

**Issue by youngm** [resilience4j/resilience4j#1855]
Resilience4j version: 2.0.2

Java version: 17

If I interrupt a thread waiting to retry using an `executeCheckedSupplier()` the below NPE is thrown.

```
java.lang.NullPointerException: Cannot throw exception because the return value of "java.util.concurrent.atomic.AtomicReference.get()" is null
	at io.github.resilience4j.retry.internal.RetryImpl$ContextImpl.waitIntervalAfterFailure(RetryImpl.java:257)
	at io.github.resilience4j.retry.internal.RetryImpl$ContextImpl.throwOrSleepAfterException(RetryImpl.java:232)
	at io.github.resilience4j.retry.internal.RetryImpl$ContextImpl.onError(RetryImpl.java:203)
	at io.github.resilience4j.retry.Retry.lambda$decorateCheckedSupplier$1(Retry.java:142)
	at io.github.resilience4j.retry.Retry.executeCheckedSupplier(Retry.java:350)
	at InterruptTest.lambda$0(InterruptTest.java:28)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

This test case works perfectly with resilience4j 1.7.0.
```
  @Test
  void test() throws Exception {
    var started = new CountDownLatch(1);
    var done = new CountDownLatch(1);
    var error = new AtomicReference<Throwable>();
    var thread =
        new Thread(
            () -> {
              try {
                Retry.of(
                        "retry",
                        RetryConfig.custom()
                            .maxAttempts(3)
                            .waitDuration(Duration.ofSeconds(10))
                            .build())
                    .executeCheckedSupplier(
                        () -> {
                          started.countDown();
                          throw new IllegalStateException("Error");
                        });
              } catch (Throwable e) {
                error.set(e);
                done.countDown();
              }
            });
    thread.start();
    assertThat(started.await(1, TimeUnit.SECONDS)).isTrue();
    assertThat(done.await(2, TimeUnit.SECONDS)).isFalse();
    thread.interrupt();
    assertThat(done.await(1, TimeUnit.SECONDS)).isTrue();
    error.get().printStackTrace();
    assertThat(error)
        .hasValueSatisfying(e -> assertThat(e).isInstanceOf(InterruptedException.class));
  }
```

**Comment by youngm** [comment]
It appears that behavior has changed with these methods since 1.x to throw the last thrown exception on interrupt instead of an InterruptedException. I think that behavior is fine for `executeSupplier()` (except the interrupted flag should be set: https://github.com/resilience4j/resilience4j/issues/1856). For `executeCheckedSupplier()` I'd be fine with the same behavior (as long as interrupted flag is set) or throwing InterruptedException either one.

**Comment by RobWin** [comment]
Hi,
would you like to create a PR to fix it?

**Comment by youngm** [comment]
I would like to but don't have any time. If it is still open when I have time perhaps I'll take a look.

**Comment by youngm** [comment]
@RobWin So, it had a little time and it appears this was already fixed! (with bb639b7d231ea6cf7e5a7a598c5749dddb3a30fc I believe) So, I'm closing this one. I did however submit a PR for https://github.com/resilience4j/resilience4j/issues/1856
